### PR TITLE
Add support for skipInitialPoll parameter (FF-2084)

### PR DESCRIPTION
**Pull request by leoromanovsky**
---
labels: mergeable
---
[//]: #  (Link to the issue corresponding to this chunk of work)
Fixes: #__issue__

## Motivation and Context
[//]: #  (Why is this change required? What problem does it solve?)

Customers can elect to put the client SDK into "offline" mode - one of the configurations they need to enable is preventing the SDK from performing a CDN fetch during the `init` method.

Subsequent calls to the CDN will be blocked by setting the `isExpired` function in the persistence store implementations.

## Description
[//]: # (Describe your changes in detail)

* adds support for js-commons `skipInitialPoll`

## How has this been tested?
[//]: # (Please describe in detail how you tested your changes)

[//]: # (OPTIONAL)
[//]: # (Add one or multiple labels: enhancement, refactoring, bugfix)

### CODE REVIEW

File: src/index.spec.ts [hunk 1]
```diff
@@ -507,6 +507,28 @@ describe('initialization options', () => {
     expect(client.getStringAssignment(flagKey, 'subject', {}, 'default-value')).toBe('control');
   });
 
+  it('skips initial poll', async () => {
```
Comment (aarsilv) [inline src/index.spec.ts:510]: 🙌

File: src/index.spec.ts [hunk 2]
```diff
@@ -507,6 +507,28 @@ describe('initialization options', () => {
...
+      });
+    }) as jest.Mock;
+
+    await init({
+      apiKey,
+      baseUrl,
+      assignmentLogger: mockLogger,
+      skipInitialPoll: true,
+    });
+    expect(callCount).toBe(0);
```
Comment (aarsilv) [inline src/index.spec.ts:529]: 👍

File: src/index.ts [hunk 3]
```diff
@@ -64,6 +64,11 @@ export interface IClientConfig {
    * backoff. (Default: 7)
    */
   numPollRequestRetries?: number;
+
+  /**
+   * Skip the first poll for new configurations during initialization. (default: false)
+   */
+  skipInitialPoll?: boolean;
```
Comment (aarsilv) [inline src/index.ts:71]: 🔥

File: src/index.ts [hunk 4]
```diff
@@ -160,6 +165,7 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
       pollAfterSuccessfulInitialization: config.pollAfterSuccessfulInitialization ?? false,
       pollAfterFailedInitialization: config.pollAfterFailedInitialization ?? false,
       throwOnFailedInitialization: true, // always use true here as underlying instance fetch is surrounded by try/catch
+      skipInitialPoll: config.skipInitialPoll ?? false,
```
Comment (aarsilv) [inline src/index.ts:168]: 👍

**Review by aarsilv — APPROVED** [review]
📈

**Review by ploomiz — APPROVED** [review]

**Review by sameerank — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (sameerank) [reply src/index.ts:71]: Just a thought: "poll" implies a request made repeatedly at intervals. `skipInitialRequest`?

Comment (leoromanovsky) [reply src/index.ts:71]: Good idea - I was being too pedantic with the mechanics of the internal poller. For customers they are not going to know that and the naming you suggested is more clear 👍 👍

**Pull request merged**